src/test/scala/edu/ie3/simona/model/grid/SystemComponentSpec.scala

```diff
@@ -103,12 +103,10 @@ class SystemComponentSpec extends UnitSpec with DefaultTestData {
         ),
       )
 
-      for ((operationStart, operationEnd, expected) <- testCases) {
+      testCases.forall { case (operationStart, operationEnd, expected) =>
```

Comment:
Please also use test cases with `forAll(testCases)` here. For this to work, the test cases have to be defined as a `Table`.